framework/src/bundle/BundlePrivate.cpp

```diff
@@ -405,7 +405,7 @@ void BundlePrivate::FinalizeActivation(LockType& l)
 
 void BundlePrivate::Uninstall()
 {
-  if (!IsSharedLibrary(GetLocation()))
+  if (BundleUtils::GetExecutablePath() == GetLocation())
```

Comment:
While code reviewing with Krish, I realized that either this throw is no longer needed OR the condition needs to also check for statically linked bundles inside of shared libraries.

Thinking about this, I think we should no longer throw since we removed the throw if a client installs a separate EXE as a bundle. This would provide symmetrical behavior between installing and uninstalling bundles (i.e. they don't throw).